arbnode/batch_poster.go

```diff
@@ -323,32 +331,52 @@ func AccessList(opts *AccessListOpts) types.AccessList {
 				// IMPLEMENTATION_SLOT from OpenZeppelin,  keccak-256 hash
 				// of "eip1967.proxy.implementation" subtracted by 1.
 				common.HexToHash("0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"),
-				// TODO: add dynamic storage slots for:
-				// - sequencerInboxAccs[sequencerInboxAccs.length - 1];
-				// - sequencerInboxAccs.push(...);
-				// - delayedInboxAccs[afterDelayedMessagesRead - 1];
 				// These below may change when transaction is actually executed:
 				// - delayedInboxAccs[delayedInboxAccs.length - 1]
 				// - delayedInboxAccs.push(...);
 			},
 		},
 	}
+
+	for _, v := range []struct{ slotIdx, val int }{
+		{7, opts.SequencerInboxAccs - 1},       // - sequencerInboxAccs[sequencerInboxAccs.length - 1]; (keccak256(7, sequencerInboxAccs.length - 1))
+		{7, opts.SequencerInboxAccs},           // - sequencerInboxAccs.push(...); (keccak256(7, sequencerInboxAccs.length))
+		{6, opts.AfterDelayedMessagesRead - 1}, // - delayedInboxAccs[afterDelayedMessagesRead - 1]; (keccak256(6, afterDelayedMessagesRead - 1))
+	} {
+
+		sia, err := intToBytes(v.val)
+		if err != nil {
+			log.Error("Error converting sequencer inbox accs to bytes", "err", err, "sequencerInboxAccs", opts.SequencerInboxAccs)
+			continue
+		}
+		sb := arbutil.SumBytes(arbutil.PaddedKeccak256([]byte{byte(v.slotIdx)}), sia)
+		l[1].StorageKeys = append(l[1].StorageKeys, common.Hash(sb))
+	}
+
 	if (opts.GasRefunderAddr != common.Address{}) {
 		l = append(l, types.AccessTuple{
 			Address: opts.GasRefunderAddr,
 			StorageKeys: []common.Hash{
 				common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000004"), // CommonParameters.{maxRefundeeBalance, extraGasMargin, calldataCost, maxGasTip}
 				common.HexToHash("0x0000000000000000000000000000000000000000000000000000000000000005"), // CommonParameters.{maxGasCost, maxSingleGasUsage}
-				//	allowedContracts[msg.sender]; for mainnet it's: "0x7686888b19bb7b75e46bb1aa328b65150743f4899443d722f0adf8e252ccda41".
-				common.Hash(arbutil.StorageSlotAddress(opts.SequencerInboxAddr.Bytes(), []byte{1})),
+				// allowedContracts[msg.sender]; for mainnet it's: "0x7686888b19bb7b75e46bb1aa328b65150743f4899443d722f0adf8e252ccda41".
+				common.Hash(arbutil.PaddedKeccak256(opts.SequencerInboxAddr.Bytes(), []byte{1})),
 				// allowedRefundees[refundee]; for mainnet it's: "0xe85fd79f89ff278fc57d40aecb7947873df9f0beac531c8f71a98f630e1eab62".
-				common.Hash(arbutil.StorageSlotAddress(opts.DataPosterAddr.Bytes(), []byte{2})),
+				common.Hash(arbutil.PaddedKeccak256(opts.DataPosterAddr.Bytes(), []byte{2})),
 			},
 		})
 	}
 	return l
 }
 
+func intToBytes(val int) ([]byte, error) {
```

Comment:
Instead of intToBytes if we do 

`big.NewInt(val).Bytes()
`
it should have same effect right? No strong preference.

Reply:
Ah nice that returns big-endian slice. Changed to that.